Set the correct executable.
Remove category "Settings"
Add Ubuntu specific gettext Domain.
Index: gparted-0.8.1/gparted.desktop.in.in
===================================================================
--- gparted-0.8.1.orig/gparted.desktop.in.in	2011-08-10 16:20:22.138993940 -0400
+++ gparted-0.8.1/gparted.desktop.in.in	2011-08-10 16:20:44.438993933 -0400
@@ -3,9 +3,10 @@
 _GenericName=Partition Editor
 _X-GNOME-FullName=GParted Partition Editor
 _Comment=Create, reorganize, and delete partitions
-Exec=@gksuprog@ @installdir@/gparted %f
+Exec=gparted-pkexec
 Icon=gparted
 Terminal=false
 Type=Application
-Categories=GNOME;System;Filesystem;
+Categories=GNOME;System;Filesystem;Settings;
 StartupNotify=true
+X-Ubuntu-Gettext-Domain=gparted
